### PR TITLE
use parts crates for rewards, show rewards in ui

### LINKED ISSUE #17372: Expedition rewards changes
The window should say what the rewards are (just generate it in advance or determine the RNG in advance) so people know if something gives a cloner board or whatever.

Also the rewards should probably be scaled up a bit. My current estimate is 2 missions getting done an hour so maybe give out like 10 rewards or something (though 10 T3 parts for 30 minutes work might still be a bit low idk).

Or at least maybe scale some of the lesser ones by count (like 1 RnD server board is probably worth 5 T4 parts idk).

@deltanedas

### PR DESCRIPTION
**Pull request by deltanedas**
## About the PR
fixes #17372

**Media**

(old, no longer has loot section)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- tweak: When taking a mission you can now see what its rewards will be.

### CODE REVIEW

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Dump the loot bit.

File: Content.Server/Salvage/Expeditions/SalvageExpeditionComponent.cs [hunk 1]
```diff
@@ -56,6 +56,12 @@ public sealed class SalvageExpeditionComponent : Component
     /// </summary>
     [ViewVariables(VVAccess.ReadWrite), DataField("difficulty")]
     public DifficultyRating Difficulty;
+
+    /// <summary>
+    /// List of items to order on mission completion
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    public List<string> Rewards = default!;
```
Comment (metalgearsloth) [inline Content.Server/Salvage/Expeditions/SalvageExpeditionComponent.cs:64]: Datafield + customtypeserializer.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by deltanedas** [comment]
dump it entirely or in ui only

**Comment by metalgearsloth** [comment]
both if possible

**Comment by deltanedas** [comment]
i think the rest of the stuff in C.Sh.Procedural.Loot is needed for ores so i left it in

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**